config/config-package.py

```diff
@@ -40,6 +41,30 @@ def copy_with_meta(template_name, destination, config_type, **kw):
         f_.write(template.render(config_type=config_type, **kw))
 
 
+class TomlArraySeparatorEncoderWithNewline(toml.TomlArraySeparatorEncoder):
+    """Special version indenting the first element of and array.
+
+    Ignores the separator argument to the constructor.
+    """
+
+    def dump_list(self, v):
+        t = []
+        retval = "[\n    "  # changed from original
+        for u in v:
+            t.append(self.dump_value(u))
+        while t != []:
+            s = []
+            for u in t:
+                if isinstance(u, list):
+                    for r in u:
+                        s.append(r)
+                else:
+                    retval += u + ",\n    "  # changed from original
+            t = s
+        retval += "]"
+        return retval
```

Comment:
If we have to subclass the config file parser/formatter to make it do what we want, why don't we subclass ConfigParser instead to allow leading indentation, like zc.buildout does?

Reply:
I only subclassed the encoder to make the result nicer.
Using the default encoder of `toml` dumps lists as:

`foo = [ 'bar', 'baz']`

which is not convenient for longer lists.

Using the base class of my inherited encoder I was able to get it to

```
foo = [ 'bar',
    'baz'
    ]
```

which not completely convinced me. I am going to create a PR for the `toml` package to get my changes upstream.

Subclassing ConfigParser is not as easy as it seems to be. Parsing an ini file is a rather complex process – I did not imagine it being that hard. I tried both `iniparse` and `ConfigUpdater` to work around the problems I had with `ConfigParser`. But both libraries have their individual problems so I decided it might be time to switch to another config file format.

It is not only about keeping the indentation but also about keeping the comments. So they get written to the next config file (e. g. `setup.cfg`).
I am not sure if `zc.buildout` actually keeps the indentation when reading the config file. (Even `ConfigParser` allows to have indentation for multi line values. But it removing the indentation is a feature to prevent values with leading whitespace. If I remember correctly `zc.buildout` does the same.)